### ISSUE TITLE
Problem with Puppeteer and app in nodejs

### ISSUE THREAD

**Issue by Sencillaweb** [Rob--W/node-xvfb#14]
Hi Rob
Sorry for my English
I have been working on this issue for two days and I do not leave, to see if you can help me or give me some clues to follow.
On a VPS with Centos8, I have an application running and scraping with Puppeteer every 15 minutes. It works well.
Now I need to get data from a web that throws me if I don't make the browser visible, but since the VPS doesn't have a screen, I had thought of Xvfb for this purpose.
However, I always get following error: 
Error: spawn Xvfb ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:240:19)
    at onErrorNT (internal/child_process.js:415:16)
    at process._tickCallback (internal/process/next_tick.js:63:19)
  errno: 'ENOENT',
  code: 'ENOENT',
  syscall: 'spawn Xvfb',
  path: 'Xvfb',
  spawnargs: [ ':99', '-screen', '0', '1920x1080x24', '-ac' ] }

My app is installed in a folder in which only the Puppeteer module, in node_modules, is installed as an extra
Now i do
npm install xvfb
without errors

Start this part of the application with:
//load puppeteer and xvfb
const puppeteer = require('puppeteer');
const Xvfb = require('xvfb');
const count = 80;
let xvfb, browser, page;
//routine extract data
(async () => 
{
		xvfb = new Xvfb({silent: true, xvfb_args: ["-screen", "0", '1920x1080x24', "-ac"]});
		xvfb.start((err)=>{if (err) console.error(err)})
		browser = await puppeteer.launch({headless: false,defaultViewport: null,args: ['--no-sandbox', '--start-fullscreen', '--display='+xvfb._display]});
        page = await browser.newPage();
     ..................
........................
...............
		await browser.close(); 
		xvfb.stop();
})();

I have tried thousands of things, keeping track of errors, changing the index.js to cypress and others, and reading all the documentation on 'spawn'. 
In addition to keeping track of all permissions, locations, paths, etc, etc ..
I can't find why spawn can't find 'Xvfb' which has just been exported "module.exports = 'Xvfb'"
If you can give me any indication I would be very grateful
Greetings and thanks

**Comment by Rob--W** [comment]
`ENOENT` means that the program cannot be found in the `PATH`. You need to install Xvfb. If it is already installed, check that it can be found in one of the locations from the `PATH` environment variable.

I haven't tried, but according to https://serverdiary.com/linux/how-to-install-xvfb-and-create-systemd-xvfb-service-on-centos-7-8-or-rhel-7-8/ `, the package is called `xorg-x11-server-Xvfb` .

Side note, an alternative to Xvfb is to use headless mode in Puppeteer.

**Comment by Sencillaweb** [comment]
Hi
Thanks for your reply Rob.
I inform you that on a server with a screen, neither scrap works in several ways, without Xvfb and with Xvfb. With puppeteer only or with puppeteer + xvfb.

But on a server without a screen, .... bufff

Yes, Xvfb installed correctly, with the paths ok, I even tell it the path directly in several ways, I also check the path before the call
 "spawn ('Xvfb' .......
and it is correct but still giving the same error.
Since last night I have been fighting with the xorg-x11-server-xvfb.
Multiple errors at the beginning, but I can already connect with the destination web.
This is costing me a lot, although I think I am already on the right track.
Let's see if I have it for today.
I've already tried the headless puppeteer in every possible way, and with the chromium arguments analyzed and changed in several possible ways as well, as studied at https://peter.sh/experiments/chromium-command-line-switches/

I have also installed Xvfb as a daemon, and it seems like it might work but still can't.

In summary:
On a Centos8 server WITH PHYSICAL SCREEN, scrap works, with puppeteer only, with puppeteer + xvfb and with puppeteer + xvfb daemon, even with puppetter + xvfb server from xorg

But, in a server with the same characteristics WITHOUT PHYSICAL SCREEN, I cannot scrape even if it manages to connect to the destination web.

It occurs to me:
I think the destination web defends itself against all this. He gets to know what I'm trying to do and blocks me somehow. And he does it very well.
Do you think the latter is possible?

In the battle I am ...
Thanks again.
Best regards.
